### PR TITLE
Initial Pearl Powder Diffraction Clean Up

### LINKED ISSUE #16460: Start a bit of cleaning up of the PEARL scripts
These scripts need a bit of refactoring before being used for other instruments/users.
First, get more demanding system tests in place:
- [ ] #16459 

Once this is done, the first obvious step is to attack code repetition / loops of algorithm calls or blocks of code, like for example:
- https://github.com/mantidproject/mantid/blob/master/scripts/PearlPowderISIS/pearl_routines.py#L750-L764
- https://github.com/mantidproject/mantid/blob/master/scripts/PearlPowderISIS/pearl_routines.py#L783-L786
- https://github.com/mantidproject/mantid/blob/master/scripts/PearlPowderISIS/pearl_routines.py#L937-L964
- https://github.com/mantidproject/mantid/blob/master/scripts/PearlPowderISIS/pearl_routines.py#L937-L964
- https://github.com/mantidproject/mantid/blob/master/scripts/PearlPowderISIS/pearl_routines.py#L937-L964
- https://github.com/mantidproject/mantid/blob/master/scripts/PearlPowderISIS/pearl_routines.py#L937-L964

A second step would be to have a close look at those "identical or nearly identical" branches for different instrument versions, for example:
- https://github.com/mantidproject/mantid/blob/master/scripts/PearlPowderISIS/pearl_routines.py#L1176-L1213
- https://github.com/mantidproject/mantid/blob/master/scripts/PearlPowderISIS/pearl_routines.py#L1221-L1255

### PR DESCRIPTION
**Pull request by shahrozahmed1**
**Description of work.**
- [x] Merged #16614
- [x] Re-run system test build

This PR cleans up the legacy Pearl Powder Diffraction script to an extend, the script contains repetition / loops of algorithms calls which were not required and been resolved. `Focus v1` and `Focus v2` now utilises the same function.

**To test:**
Check that all system test pass
Code changes look acceptable
Run some of the scripts provided by Craig locally
Check if any further refactoring that could be made here

Fixes #16460 
_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/PearlPowderISIS/pearl_routines.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# pylint: disable=anomalous-backslash-in-string, global-variable-undefined, global-variable-not-assigned, too-many-lines
```
Comment (FedeMPouzols) [inline scripts/PearlPowderISIS/pearl_routines.py]: Nice to get rid of some of these.

File: scripts/PearlPowderISIS/pearl_routines.py [hunk 2]
```diff
@@ -341,7 +340,14 @@ def PEARL_focus(number, ext="raw", fmode="trans", ttmode="TT70", atten=True, van
...
 
-def PEARL_focus_v1(number, ext="raw", fmode="trans", ttmode="TT70", atten=True, van_norm=True, debug=False):
+def pearl_run_focus(number, ext="raw", fmode="trans", ttmode="TT70", atten=True, van_norm=True, debug=False,
+                    focus_mode=1):
+    alg_range = 12
+    save_range = 3
+    if focus_mode == 2:
+        alg_range = 14
+        save_range = 5
+
```
Comment (FedeMPouzols) [inline scripts/PearlPowderISIS/pearl_routines.py]: Aha interesing, so this was all the difference between focus_v1 and focus_v2?

@shahrozahmed1: I see this is "in progress". I skimmed through the code changes and it looks good. Let me know when #16459 and this are ready for testing.

File: scripts/PearlPowderISIS/pearl_routines.py [hunk 3]
```diff
@@ -936,38 +687,18 @@ def PEARL_createvan(van, empty, ext="raw", fmode="all", ttmode="TT88",
...
-        print "About to strip Work=12 (again)"
-        StripPeaks(InputWorkspace="vanstrip", OutputWorkspace="vanstrip", FWHM=100, Tolerance=10, WorkspaceIndex=12)
-        print "About to strip Work=13 (again)"
-        StripPeaks(InputWorkspace="vanstrip", OutputWorkspace="vanstrip", FWHM=60, Tolerance=10, WorkspaceIndex=13)
+        print("About to strip work=12 and work=13 twice")
+        for i in range(0, 2):
+            print "About to strip Work=12"
+            StripPeaks(InputWorkspace="vanstrip", OutputWorkspace="vanstrip", FWHM=100, Tolerance=10, WorkspaceIndex=12)
+            print "About to strip Work=13"
+            StripPeaks(InputWorkspace="vanstrip", OutputWorkspace="vanstrip", FWHM=60, Tolerance=10, WorkspaceIndex=13)
```
Comment (FedeMPouzols) [inline scripts/PearlPowderISIS/pearl_routines.py]: All looks like a great code improvement. I only spotted this loop that seems to be repeating the same couple of `StripPeaks` two times?

Comment (shahrozahmed1) [inline scripts/PearlPowderISIS/pearl_routines.py]: @FedeMPouzols Yes apparently `12` and `13` are meant to be stripped twice; as the comment above says *run twice on low angle as peaks are very broad*

Comment (FedeMPouzols) [inline scripts/PearlPowderISIS/pearl_routines.py]: Ah I see, it makes sense.

**Comment by FedeMPouzols** [comment]
This is a huge code improvement in the PEARL scripts.
- Removes lots of redundant code (unrolled loops, a nearly duplicate function, etc.) and some pylint issues.
- Locally, the script behaves well with commands from the documentation examples
- Passes the system tests, including all the new checks added in #16614 in a re-run after that branch was merged into master.

:shipit:

**Pull request merged**